### PR TITLE
Add sass-loader to webpack configs

### PR DESCRIPTION
**Pull request by thundron**
Change-type: patch
Changelog-entry: Add sass-loader to webpack configs
Signed-off-by: Lorenzo Alberto Maria Ambrosi <lorenzoa@balena.io>

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -19,63 +19,49 @@
...
       // eslint-disable-next-line lodash/prefer-lodash-method
       const absoluteContext = path.resolve(context)
       const absoluteNodeModules = path.resolve('node_modules')
 
+      // if (/\/fonts/i.test(request)) {
+      //   const output = path.join(__dirname, 'generated')
+      //   const fontName = _.last(_.split(request, '/'))
+      //   const fontPath = path.join(output, 'fonts', fontName)
+      //   return callback(null, `commonjs ${fontPath}`)
+      // }
```
Comment (zvin) [inline webpack.config.js]: commented code

File: webpack.config.js [hunk 2]
```diff
@@ -19,63 +19,49 @@
...
   },
+  target: 'electron-renderer',
+  optimization: {
+    // Minification breaks angular.
+    minimize: false
+  },
   externals: [
     nodeExternals(),
     (context, request, callback) => {
       // eslint-disable-next-line lodash/prefer-lodash-method
```
Comment (zvin) [inline webpack.config.js]: looks useless, is there a lodash equivalent for `path.resolve`?

File: webpack.config.js [hunk 3]
```diff
@@ -19,63 +19,49 @@
...
-  ]
+  ],
+  output: {
+    path: path.join(__dirname, 'generated'),
+    filename: '[name].js'
+  }
 }
 
-const guiConfig = _.assign({
+const guiConfig = _.assign(_.cloneDeep(commonConfig), {
```
Comment (zvin) [inline webpack.config.js]: You can use `_.assign({}, commonConfig, { other: stuff })` and avoid the `cloneDeep`.

File: webpack.config.js [hunk 4]
```diff
@@ -97,20 +83,71 @@ const guiConfig = _.assign({
...
+  plugins: [
+    new MiniCssExtractPlugin({
+      filename: '[name].css',
+      chunkFilename: '[id].css'
+    })
+  ]
+})
 
-const etcherConfig = _.assign({
+const etcherConfig = _.assign(_.cloneDeep(commonConfig), {
```
Comment (zvin) [inline webpack.config.js]: You can use `_.assign({}, commonConfig, { other: stuff })` and avoid the cloneDeep.

**Comment by lurch** [comment]
I guess this PR should also be making changes to the sass-related stuff in `Makefile` ?

**Comment by thundron** [comment]
> I guess this PR should also be making changes to the sass-related stuff in `Makefile` ?

@lurch Exactly, it's still a WIP because fonts paths are giving me nightmares right now

File: webpack.config.js (hunk 2 above)
Comment (thundron) [reply webpack.config.js]: I don't think so, removed

**Comment by thundron** [comment]
Depends on #2795 to be merged before (refining last things)

**Pull request merged**